File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem41_SearchInABinarySearchTreeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem41_SearchInABinarySearchTreeTest {
    private final Problem41_SearchInABinarySearchTree test = new Problem41_SearchInABinarySearchTree();

    @Test
    void searchBST() {
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(2);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(3);
        root.right = new TreeNode(7);

        assertEquals(1, test.searchBST(root, 2).left.val);
    }
}

File: JavaScript/leetcode75/src/test/Problem41_SearchInABinarySearchTree.test.js
const searchBST = require("../main/Problem41_SearchInABinarySearchTree");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return the sub tree with the root node value", () => {
  let root = new TreeNode(4);
  root.left = new TreeNode(2);
  root.left.left = new TreeNode(1);
  root.left.right = new TreeNode(3);
  root.right = new TreeNode(7);
  expect(searchBST(root, 2)?.left?.val).toBe(1);
});
